lib/timing.mjs

```diff
@@ -0,0 +1,125 @@
+import process from 'node:process';
+import { table } from 'table';
+
+// Inspired by ESLint's timing.js
+// https://github.com/eslint/eslint/blob/09bc2a88c00aa9a93c7de505795fc4e85b2e6357/lib/linter/timing.js
+
+/**
+ * Start time measurement.
+ * @returns {bigint} Variable for tracking time in nanoseconds.
+ */
+function startTime() {
+	return process.hrtime.bigint();
+}
+
+/**
+ * End time measurement.
+ * @param {bigint} start Variable for tracking time in nanoseconds.
+ * @returns {number} The measured time in milliseconds.
+ */
+function endTime(start) {
+	const diff = process.hrtime.bigint() - start;
+
+	return Number(diff) / 1e6;
+}
+
+const enabled = Boolean(process.env.TIMING);
+const HEADERS = ['Rule', 'Time (ms)', 'Relative'];
+
+/**
+ * Decide how many rules to show in the output list.
+ * @returns {number} The number of rules to show.
+ */
+function getListSize() {
+	const TIMING = process.env.TIMING;
+	const MINIMUM_SIZE = 10;
+
+	if (typeof TIMING !== 'string') {
```

Comment:
nit
style

```suggestion
	if (typeof TIMING === 'undefined') {
```

I think this would be clearer because an environment variable is always `string | undefined`.
Either way it's fine.

Reply:
Fixed it.
https://github.com/stylelint/stylelint/pull/8108/commits/ebbdd0deb70cd744dd77f8817854ef94e0152e5e